singer_sdk/connectors/sql.py

```diff
@@ -959,10 +959,18 @@ def _adapt_column_type(
         Raises:
             NotImplementedError: if altering columns is not supported.
         """
+        collation_removed = False
         current_type: sqlalchemy.types.TypeEngine = self._get_column_type(
             full_table_name, column_name
         )
 
+        # remove collation if present and save it
+        if hasattr(current_type, "collation"):
```

Comment:
Missing coverage here may point to a useful refactor that can be unit-tested:

Wdyt about moving this block to a helper function that takes a type object and returns the collation, or `None` if the collation is falsy:

```python
def remove_collation(column_type: TypeEngine) -> "Collation" | None:  # don't know the actual type of a collation
    if hasattr(current_type, "collation") and current_type.collation:
        current_type_collation = current_type.collation
        setattr(current_type, "collation", None)
        return current_type_collation
    return None

def update_collation(column_type: TypeEngine, collation: "Collation" | None) -> None:
    """Replace column collation if it has one."""
    if hasattr(column_type, "collation") and collation:
        setattr(column_type, "collation", collation)
```

Later we can check if the collation is not `None` to set it in the new type

```python
        current_type_collation = remove_collation(current_type)

        ...

        update_collation(compatible_sql_type, current_type_collation)
```

Reply:
I think this looks great. I will work on getting these two functions in place and tests for them.

Reply:
Thanks @BuzzCutNorman!

Reply:
@edgarrmondragon when you have a moment, please review the changes I made.